### PR TITLE
Add full manifold model capability

### PR DESCRIPTION
**Pull request by baperry2**
Depends on #414

### CODE REVIEW

**Comment by baperry2** [comment]
This is now mostly ready, except for documentation and adding to the CI. 

There is a FlameSheet example case (`input.manifold`) that can be run by compiling with `USE_MANIFOLD = TRUE`. That case requires a pmf data file and a tabulated chemistry file that can be run using the script `get_pmfdat_and_chemtable.py` in the CMLM repo: https://github.com/NREL/cmlm/blob/development/get_pmfdat_and_ctable.py

The test case uses a 1D table based on cantera premixed flames at the same conditions as the detailed chemistry FlameSheet test.

File: Source/PeleLMeX_Setup.cpp [hunk 1]
```diff
@@ -459,6 +467,11 @@ PeleLM::readParameters()
                       "fixed_Pr or fixed_Le is true"
                    << std::endl;
   }
+  if (
+    m_use_wbar != 0 &&
```
Comment (marchdf) [inline Source/PeleLMeX_Setup.cpp]: add extra paren?

File: Source/PeleLMeX_Setup.cpp [hunk 2]
```diff
@@ -1118,6 +1131,20 @@ PeleLM::derivedSetup()
     "enstrophy", IndexType::TheCellType(), 1, pelelmex_derenstrophy,
     grow_box_by_two);
 
+#ifdef USE_MANIFOLD_EOS
+  auto& mani_data = eos_parms.host_only_parm().manfunc_par->host_parm();
+  int nmanivar = mani_data.Nvar;
```
Comment (marchdf) [inline Source/PeleLMeX_Setup.cpp]:
```suggestion
  const int nmanivar = mani_data.Nvar;
```

File: Source/PeleLMeX_K.H [hunk 3]
```diff
@@ -625,7 +735,8 @@ makeVelForce(
   const amrex::Real& rho_incomp,
   int pseudo_gravity,
   int pseudo_gravity_dir,
-  const amrex::Real& /*time*/,
+  const amrex::Real& /*time*/
+  ,
```
Comment (marchdf) [inline Source/PeleLMeX_K.H]: this is a strange diff?

**Review by marchdf — APPROVED** [review]
Left some minor comments

File: Exec/Make.PeleLMeX [hunk 4]
```diff
@@ -73,6 +73,20 @@ endif
 ifeq ($(Eos_Model),$(filter $(Eos_Model),Soave-Redlich-Kwong))
    DEFINES += -DUSE_SRK_EOS
 endif
+ifeq ($(Eos_Model),$(filter $(Eos_Model),Manifold))
+   DEFINES += -DUSE_MANIFOLD_EOS
+   DEFINES += -DUSE_MANIFOLD_EOS
```
Comment (drummerdoc) [inline Exec/Make.PeleLMeX]: why twice?

File: Exec/Make.PeleLMeX [hunk 5]
```diff
@@ -73,6 +73,20 @@ endif
...
+   DEFINES += -DMANIFOLD_DIM=$(Manifold_Dim)
+   ifeq ($(Manifold_Type),Table)
+      DEFINES += -DMANIFOLD_EOS_TYPE=1
+   else
+      ifeq ($(Manifold_Type),Network)
+         DEFINES += -DMANIFOLD_EOS_TYPE=2
+      else
+         DEFINES += -DMANIFOLD_EOS_TYPE=0
```
Comment (drummerdoc) [inline Exec/Make.PeleLMeX:89]: Wonder if, later, these can be words rather than integers.

**Review by drummerdoc — APPROVED** [review]

File: Exec/Make.PeleLMeX (hunk 4 above)
Comment (baperry2) [reply Exec/Make.PeleLMeX]: fixed.

File: Exec/Make.PeleLMeX (hunk 5 above)
Comment (baperry2) [reply Exec/Make.PeleLMeX:89]: At least the user-facing part is words for now.

**Comment by baperry2** [comment]
Going to merge this now. Will add docs and more testing in future PRs.

**Pull request merged**